### PR TITLE
Implemented result page design with dynamically generated fun fact

### PR DESCRIPTION
**Pull request by CallaJune**
## What
This pull request implements the proposed design for the result page, courtesy of Prateek Upreti. The design includes a randomly generated "fact" about the user's accomplishments, functionality I have also implemented. 

## How
* I used styled-components in order to style the result page using CSS. The design was inspired by this mock-up:
Screen Shot 2021-02-03 at 6 01 21 PM

* In order to implement the result page fact, I created a `fact` state variable in `mapping.js` which is passed into the result page. `fact` is fetched from a `getFact/[uuid]` API call, which I have implemented. 
* A small note: Calls made to fetch user stats, the number of untagged locations, and the fact were being made infinitely on the result page, and so I have used `counter` and added some logic in `mapping.js` to make sure that these function calls are made only once. While I could have done this check in `handleNextSelected`, that would result in the functions not being called when the user has no locations left to map and thus does not ever click "Next." 
* The `getFact` API is documented further in the comments in `getFact/[uuid].js`. It essentially includes arrays and dictionaries containing multiple fact queries. The API then randomly selects one and returns a fact ready to be placed on the result page. 
* Two facts have been added: the user's top country and the number of distinct countries they have mapped. Here are some sample screenshots:
Screen Shot 2021-02-03 at 5 54 45 PM
Screen Shot 2021-02-03 at 6 13 10 PM

## Tests
I have tested this throughout development to assure that it behaves as expected.
* I have run the SQL queries directly against my local Postgres database to assure they return the expected information in the required format.
* I have verified that the fact generated at the end is randomly generated by generating the result page 50 times. In my test, there were two possible facts. One appeared 23 times and the other appeared 27 times. Both facts were represented. 
* I've played the game and tested the grammar of the facts when all locations have not yet been mapped, and when all locations have been mapped. 

## Notes
This PR requires that locations loaded include country codes.

### CODE REVIEW

File: pages/mapping.js [hunk 1]
```diff
@@ -101,32 +102,43 @@ export default function mapping() {
...
 			countryCounts[question.country_code] = 1;
 		}
-		setGameStats({ country_counts: countryCounts });
+		setGameStats({ country_counts: countryCounts, mapped_count: mappedCount + 1 });
 
 		if (counter < questions.length) {
 			setCounter(counter + 1);
 			setLocationResults({ yes_count: 0, no_count: 0, maybe_count: 0, total_count: 0});
+	  	} else {
+	  		fetchUserStats();
```
Comment (lacabra) [inline pages/mapping.js:117]: Based on your PR comments, I believe you didn't intend to keep these 3 lines here

File: pages/mapping.js [hunk 2]
```diff
@@ -101,32 +102,43 @@ export default function mapping() {
...
 	  	}
 	}
 
 	async function fetchUntaggedLocationsCount() {
 		// Get number of untagged locations
-		const result = await fetch(`/api/getUntaggedLocationsCount/${cookies.uuid}`);
+		const result = await fetch(`/api/getUntaggedLocationsCount/${cookies.uuid}`)
 		const response = await result.json();
 		setUntaggedLocationsCount(await response.count);
 	};
```
Comment (lacabra) [inline pages/mapping.js:128]: We had an offline conversation about this, but we got the line numbers mixed up. This is the line where the `;` should be removed

File: pages/mapping.js [hunk 3]
```diff
@@ -101,32 +102,43 @@ export default function mapping() {
...
 
 	async function fetchUserStats() {
-		const result = await fetch(`/api/getUserStats/${cookies.uuid}`);
+		const result = await fetch(`/api/getUserStats/${cookies.uuid}`)
 		const response = await result.json();
 		setUserStats(await response);
 	};
```
Comment (lacabra) [inline pages/mapping.js:134]: same here, no `;` is customary

**Review by lacabra — CHANGES_REQUESTED** [review]
Thanks @CallaJune, this looks very well overall. Made some fairly small observations.

File: pages/mapping.js (hunk 1 above)
Comment (CallaJune) [reply pages/mapping.js:117]: Yes. Thanks.

**Review by lacabra — APPROVED** [review]
Looks good 🙌 Thank you @CallaJune!

**Pull request merged**